pkg/services/hcloud/remediation/remediation.go

```diff
@@ -58,8 +58,10 @@ func (s *Service) Reconcile(ctx context.Context) (res reconcile.Result, err erro
 		s.scope.HCloudRemediation.Status.Phase = infrav1.PhaseDeleting
 
 		if err := s.setOwnerRemediatedCondition(ctx); err != nil {
+			record.Warn(s.scope.HCloudRemediation, "FailedSettingConditionOnMachine", err.Error())
 			return res, fmt.Errorf("failed to set conditions on CAPI machine: %w", err)
 		}
+		record.Warn(s.scope.HCloudRemediation, "FailedToFindServer", "Server doesn't exist")
```

Comment:
make a comment like the one above with "exit remediation because bare metal server does not exist"
-> FailedToFindServer is not very helpful if you see only that event